ambry-account/src/main/java/com/github/ambry/account/HelixAccountServiceFactory.java

```diff
@@ -47,27 +48,27 @@
    * Constructor.
    * @param verifiableProperties The properties to get a {@link HelixAccountService} instance. Cannot be {@code null}.
    * @param metricRegistry The {@link MetricRegistry} for metrics tracking. Cannot be {@code null}.
-   * @param notifier The {@link Notifier} used to get a {@link HelixAccountService}. Can be {@code null}.
    */
-  public HelixAccountServiceFactory(VerifiableProperties verifiableProperties, MetricRegistry metricRegistry,
-      Notifier<String> notifier) {
+  public HelixAccountServiceFactory(VerifiableProperties verifiableProperties, MetricRegistry metricRegistry) {
     storeConfig = new HelixPropertyStoreConfig(verifiableProperties);
     accountServiceConfig = new HelixAccountServiceConfig(verifiableProperties);
     accountServiceMetrics = new AccountServiceMetrics(metricRegistry);
-    this.notifier = notifier;
+    notifier =
+        accountServiceConfig.zkClientConnectString.equals(HelixAccountServiceConfig.INVALID_ZK_CLIENT_CONNECT_STRING)
```

Comment:
I don't think the `if` case is required here since it is assumed that `accountServiceConfig.zkClientConnectString` is not invalid (it is used in `getAccountService()`). What do you think?

Reply:
You also say it needs to be valid when using `HelixAccountService` in the config doc

Reply:
The `if` case is not necessary for production code, but needed for unit tests.  In `HelixAccountServiceTest` and `HelixNotifierTest`, the `ZK_CONNECT_STRING` is a dummy string (I purposely use empty string as invalid zkClientConnectString). In this case,  notifier is null and will not try to connect zookeeper(which definitely will fail and timeout).